src/lib.rs

```diff
@@ -44,17 +43,24 @@ impl Playpen {
         (compiler, output)
     }
 
-    pub fn evaluate(&self, branch: Branch, code: String) -> io::Result<(ExitStatus, String, String)> {
-        let (status, raw_output) = self.exec(branch, "/usr/local/bin/evaluate.sh", vec![], code)?;
-        let (compiler, output) = Self::parse_output(&raw_output);
-        Ok((status, compiler, output))
+    pub async fn evaluate(branch: Branch, code: String) -> Result<(RunResult, String, String)> {
+        let result = Self::exec(branch, "/usr/local/bin/evaluate.sh", vec![], code).await?;
```

Comment:
Just curious since I don't know too much about async rust: why prefer this style over taking `self`? Is it related to making the functions `async`, or is it something else?